src/odbc.cpp

```diff
@@ -238,36 +248,31 @@ ODBCError* ODBCAsyncWorker::GetODBCErrors
 
   ODBCError *odbcErrors = new ODBCError[statusRecCount]();
   this->errorCount = statusRecCount;
-
   for (SQLSMALLINT i = 0; i < statusRecCount; i++) {
 
     ODBCError error;
 
-
-    return_code = SQLGetDiagRec(
+    return_code = SQLGetDiagRec
+    (
       handleType,                 // HandleType
       handle,                     // Handle
       i + 1,                      // RecNumber
-      errorSQLState,              // SQLState
-      &nativeError,               // NativeErrorPtr
-      errorMessage,               // MessageText
+      error.state,                // SQLState
+      &error.code,                // NativeErrorPtr
+      error_message,               // MessageText
       ERROR_MESSAGE_BUFFER_CHARS, // BufferLength
-      &textLength                 // TextLengthPtr
+      &error_message_length       // TextLengthPtr
     );
 
     if (SQL_SUCCEEDED(return_code)) {
-      error.state = errorSQLState;
-      error.code = nativeError;
```

Comment:
The new code isn't safe, unfortunately, since the returned field contains the "total number of characters (excluding the number of characters required for the null-termination character) _available_ to return" which could be larger than the `error_message` size.